### ISSUE TITLE
[Regression] Composite unique index adds single column unique index to the columns

### ISSUE THREAD

**Issue by alitaheri** [sequelize/sequelize#7771]
## What you are doing?

```js
const Sequelize = require('sequelize');
const sequelize = new Sequelize('db', 'user', 'pass', {
  host: 'localhost',
  port: 5432,
  dialect: 'postgres',
});

const Test = sequelize.define(
  'test', {
    unique1: Sequelize.INTEGER,
    // even unique; false doesn't override the effect
    unique2: {type: Sequelize.INTEGER, unique: false},
  }, {
    indexes: [{
      unique: true,
      fields: ['unique1', 'unique2']
    }]
  },
);

Test.sync({force: true})

```

## What do you expect to happen?
Previously ( `4.0.0-2` ) used to define **only** composite unique index. Which is the expected behavior.

## What is actually happening?
But now it defines 3 unique indices:

1. `UNIQUE (unique1)`
2. `UNIQUE (unique2)`
3. `UNIQUE (unique1, unique2)`

I think the culprit is #7196.

This is obviously a regression as my queries are now broken, because `unique1` and `unique2` should be unique in combination not in their own.

_Output, either JSON or SQL_

Constraint check errors... 😱 😱 

__Dialect:__ postgres (possibly any)
__Database version:__ 9.6
__Sequelize version:__ 4.0.0

**Comment by alitaheri** [comment]
@felixfbecker @sushantdhiman This is an ugly regression, took me 10 minutes to get to the bottom of it. I suggest we revert #7196 and try to solve the issue differently (or at least in a way that doesn't affect sync) Maybe a new property that only upsert is aware of like: `uniqueConstraintParticipant`.

**Comment by jeffmoss** [comment]
Is there some trickery I can use to get the "indexes: [{ unique: true }]" to work right in my definition? I have two sets of overlapping unique-composite indexes (for upsert to work) and so I can't make use of the unique: 'string' functionality.

**Comment by alitaheri** [comment]
@jeffmoss Right now you should drop the extra indices after sync.

**Comment by yepp4you** [comment]
Why do you have to set unique fields for each?

model.js 

this.options.indexes = this.options.indexes.map(index => {
      index = this._conformIndex(index);
      //Add unique indexes to rawAttributes
      if (index.unique && index.fields) {
        for (const field of index.fields) {
          const fieldName = typeof field === 'string' ? field : field.name || field.attribute;
          if (this.rawAttributes[fieldName]) {
            this.rawAttributes[fieldName].unique = this.attributes[fieldName].unique = index.name || index.unique;
          }
        }
      }

      return index;

**Comment by alitaheri** [comment]
@yepp4you This was a naive fix for upsert. But it broke other things.

**Comment by alitaheri** [comment]
The fix is at #7854 @felixfbecker I suggest a release after merge. We should contain this before more people are affected 😅 😅

**Comment by felixfbecker** [comment]
Every merge is an automatic release
